pkg/routing/reconciler/common/reconcile.go

```diff
@@ -100,16 +100,16 @@ func (r *GenericDeploymentReconciler) ReconcileAdapter(ctx context.Context, ab A
 // resolveSinkURL resolves the URL of a sink reference.
 func (r *GenericDeploymentReconciler) resolveSinkURL(ctx context.Context) (*apis.URL, error) {
 	router := v1alpha1.RouterFromContext(ctx)
-	sink := *router.GetSink()
+	sink := router.GetSink()
 
-	if sinkRef := &router.GetSink().Ref; *sinkRef != nil && (*sinkRef).Namespace == "" {
-		(*sinkRef).Namespace = router.GetNamespace()
+	if sinkRef := sink.Ref; sinkRef != nil && sinkRef.Namespace == "" {
+		sinkRef.Namespace = router.GetNamespace()
 	}
 
-	return r.SinkResolver.URIFromDestinationV1(ctx, sink, router)
+	return r.SinkResolver.URIFromDestinationV1(ctx, *sink, router)
```

Comment:
We store the result of `GetSink()` into a variable which we don't use in the conditional below.
With this change, we do.

Also there was a lot of unnecessary pointer magic here (I'm the original perpetrator of this crime).